corehq/apps/app_manager/views/releases.py

```diff
@@ -388,6 +402,19 @@ def _track_build_for_app_preview(domain, couch_user, app_id, message):
     })
 
 
+def _check_app_for_mobile_ucr_v1_refs(domain, app):
+    if not toggles.MOBILE_UCR.enabled(domain):
+        return
+    if app.mobile_ucr_restore_version != '2.0':
+        return _("The mobile UCR restore version for v%(app_version)s needs to be updated to V2.0") % {
+            'app_version': app.version,
+        }
+    if does_app_have_mobile_ucr_v1_refs(app):
+        return _("One or more forms for v%(app_version)s contain V1 Mobile UCR references.") % {
```

Comment:
Wondering if we should add something more here so that the user is enabled to take next steps after receiving the warning ?   
Just thinking of scenarios, I don't think any action can be taken when reverting to old versions.
As for new builds, It might not be required once we run the migration script.

Reply:
> Wondering if we should add something more here so that the user is enabled to take next steps after receiving the warning ?

That could be a good idea, however I'm not sure what else to add here without making the warning text quite long. I could add something along the lines of `One or more forms for v%(app_version)s contain V1 Mobile UCR references. Please remove all manual v1 references before making a new app release`, however this still doesn't feel very helpful in terms of giving next steps.

@ajeety4 Any thoughts/suggestions on what you think can be added here?

> I don't think any action can be taken when reverting to old versions.

When reverting to an old version a user would need to make the necessary changes to remove all v1 references and then create a new app build containing only v2 references. So long as a user doesn't attempt to release an app containing v1 refs they won't be blocked with reverting to it.

Reply:
>  When reverting to an old version a user would need to make the necessary changes to remove all v1 references and then create a new app build containing only v2 references. So long as a user doesn't attempt to release an app containing v1 refs they won't be blocked with reverting to it.

That makes sense.  Thanks for clearing this.

> this still doesn't feel very helpful in terms of giving next steps. @ajeety4 Any thoughts/suggestions on what you think can be added here?

Agreed. I think having a link to documentation with steps would have been good, however this is behind a feature flag.
I don't have a better suggestion and am good with leaving as it is.  Maybe check with Yoilswa once since he would be assisting projects to get rid of manual references and can share inputs if any.